python_modules/dagster/dagster_tests/storage_tests/branching_io_manager_tests/utils.py

```diff
@@ -86,14 +86,22 @@ def __init__(self):
 
     def handle_output(self, context: OutputContext, obj: Any):
         keys = self._keys_from_context(context)
-        for key in keys:
-            self.values[key] = obj
+        if keys is None:
+            self.values[None] = obj
```

Comment:
this pattern is odd to me - did you need it to get tests passing?

Reply:
It's a refactor of what was already happening, and it was the simplest way to get the type-checking to pass. Previously `_keys_from_context` would sometimes return `[None]` (IMO quite confusing) and we would still have `self.values[None]` set, only implicitly.